python/pyarrow/io.pxi

```diff
@@ -1053,6 +1061,14 @@ cdef class Buffer(_Weakrefable):
     def __len__(self):
         return self.size
 
+    def __repr__(self):
+        name = f"{self.__class__.__module__}.{self.__class__.__name__}"
+        return (f"{name}("
+                f"address={hex(self.address)}, "
+                f"size={self.size}, "
+                f"is_cpu={self.is_cpu}, "
+                f"is_mutable={self.is_mutable})")
```

Comment:
So, just for the record, this makes it look like the `Buffer` constructor is callable with these arguments (which it is not).
We could instead go for: `<pyarrow.Buffer address=0x...>`

@jorisvandenbossche What do you think?

Reply:
+1 on keeping the `< .. >` (instead of `()`) to not confuse it with an eval-able repr

Reply:
https://github.com/apache/arrow/pull/13921/commits/febd2efa417feba35b57ca41481b18c0992fd9c8